internal/wrapper/wrap_handler.go

```diff
@@ -46,15 +43,14 @@ func WrapHandlerWithListeners(handler interface{}, listeners ...HandlerListener)
 
 	// Return custom handler, to be called once per invocation
 	return func(ctx context.Context, msg json.RawMessage) (interface{}, error) {
+		ctx = context.WithValue(ctx, "cold_start", coldStart)
 		for _, listener := range listeners {
 			ctx = listener.HandlerStarted(ctx, msg)
 		}
-		ctx = context.WithValue(ctx, "cold_start", coldStart)
 		CurrentContext = ctx
-		submitEnhancedMetrics("invocations", ctx)
 		result, err := callHandler(ctx, msg, handler)
 		if err != nil {
-			submitEnhancedMetrics("errors", ctx)
+			ctx = context.WithValue(ctx, "error", true)
```

Comment:
Here, this is a really good pattern for storing Context keys: https://medium.com/@matryer/context-keys-in-go-5312346a868d